### DISCUSSION TITLE
Metal item uncraft

### DISCUSSION THREAD

**Discussion by ghost** [cataclysmbn/Cataclysm-BN#3298] [General]
Put here to gain feedback so that when I do this I do it right. Metal items can be smithed but not uncrafted, thus precious metal can't be recycled. I have a few ideas here.

1. It requires all forging heat to deconstruct. Tongs and crucible required since its melting it down here.
2. It requires a little heat plus either normal or fine metal sawing.
3. It requires only fine metal saw. (I don't like normal metal sawing here as forges give metal sawing 1 for no energy input)

Next: Some assumptions to be made about the uncraft. I would avoid using reversible due to taking multiple hours just to saw down a pot into usable metal.

1. The biggest size of steel will be given, as you're taking a finished item and breaking it down to reasonable size.
2. It's going to be faster but we need times. 5 minutes, or 10 minutes per lump of steel. Less for chunks and scrap.

Time is very subject to feedback and changes. I would like to nail down time assumptions before starting. This is a tedious JSON project that will take hours potentially.

**Comment by chaosvolt** [comment]
So, on this topic, I do wish we had either:
1. Being able to alter the yield of uncrafts based on item components.
2. Having JSONized ability to modify the proprieties of a reversible recipe.

The latter is done to a limited extent but is hardcoded, and it'd be really useful for shit like making sure that pistol vs. rifle rounds consistently use the correct level of pulling quality to reverse a recipe.

**Comment by ghost** [comment]
Like the clown I sometimes am, I finally noticed copper pots are uncraftable with metal sawing 1. Seems that fine metal sawing is the way to go for steel items, without any forge charges required in order to maintain internal consistency. How I've gone this many years without gathering scrap copper I don't know I guess I got lucky with other spawns. A rapid 7 minutes for 5 scrap copper and 1 scrap metal means I could consider 5 minutes per lump of steel fair.

Indeed I've gone this long too without attempting to saw down car frames, 20 lumps of steel for 45 minutes and fine metal sawing 1 so thats about 2 minutes per lump.

**Comment by Unrepentant-Atheist** [comment]
The vanilla game already has a machine to somewhat do that `t_recycler`.

```
  {
    "type": "terrain",
    "id": "t_recycler",
    "name": "metal compactor",
    "description": "A hydraulic compactor that can accept items made of various metals, and press them into basic shapes, ready for further crafting.",
    "looks_like": "t_machinery_heavy",
    "symbol": "&",
    "color": "green",
    "move_cost": 0,
    "max_volume": "2000 L",
    "flags": [ "TRANSPARENT", "REDUCE_SCENT", "PERMEABLE" ],
    "examine_action": "recycle_compactor",
    "bash": {
      "str_min": 20,
      "str_max": 150,
      "sound": "metal screeching!",
      "sound_fail": "clang!",
      "ter_set": "t_thconc_floor",
      "items": [
        { "item": "steel_lump", "count": [ 0, 2 ] },
        { "item": "steel_chunk", "count": [ 1, 4 ] },
        { "item": "scrap", "count": [ 3, 12 ] }
      ]
    }
  },
  ```

**Comment by ghost** [comment]
Likely this will be sorted with salvage actions when I test to see if new ones are still possible in JSON, and if not then a code update to get it working. The documentation is 8 years old so who knows.

The recycler doesn't count as a solution to this and was confirmed by some maintainers to not be worth using the action for.
